Sources/NIO/EventLoop.swift

```diff
@@ -770,6 +770,14 @@ extension EventLoopGroup {
     }
 }
 
+/// Specifies how `EventLoopGroup` will be created and establishes lifecycle ownership.
+public enum EventLoopGroupProvider {
+    /// `EventLoopGroup` will be provided by the user. Owner of this group is responsible for its lifecycle.
+    case shared(EventLoopGroup)
+    /// `EventLoopGroup` will be created by the client. When `syncShutdown` is called, created `EventLoopGroup` will be shut down as well.
```

Comment:
This comment needs updating to reflect the fact that this won't necessarily be used by a client nor will it  necessarily have a `syncShutdown`.

Reply:
Perhaps rephrased as:

```suggestion
    /// Create a new `EventLoopGroup` when necessary. 
    /// The library which accepts this provider takes ownership of the created event loop group, 
    /// and must ensure its proper shutdown when the library is being shut down.
```

?